src/tls/cockpit-certificate-helper.in

```diff
@@ -28,6 +28,65 @@ install_key() {
     fi
 }
 
+selfsign_sscg() {
+    sscg --quiet \
+        --lifetime "${DAYS}" \
+        --key-strength 2048 \
+        --cert-key-file "${KEYFILE}" \
+        --cert-file "${CERTFILE}" \
+        --ca-file "${CA_FILE}" \
+        --hostname "${HOSTNAME}" \
+        --organization "${MACHINE_ID}" \
+        --subject-alt-name localhost \
+        --subject-alt-name IP:127.0.0.1/255.255.255.255
+}
+
+selfsign_openssl() {
+    openssl req -x509 \
+        -days "${DAYS}" \
+        -newkey rsa:2048 \
+        -keyout "${KEYFILE}" \
+        -keyform PEM \
+        -nodes \
+        -out "${CERTFILE}" \
+        -outform PEM \
+        -subj "${MACHINE_ID:+/O=${MACHINE_ID}}/CN=${HOSTNAME}" \
+        -config - \
+        -extensions v3_req << EOF
+    [ req ]
+    req_extensions = v3_req
+    extensions = v3_req
+    distinguished_name = req_distinguished_name
+    [ req_distinguished_name ]
+    [ v3_req ]
+    subjectAltName=IP:127.0.0.1,DNS:localhost
+    basicConstraints = critical, CA:TRUE
+    keyUsage = critical, digitalSignature,cRLSign,keyCertSign,keyEncipherment,keyAgreement
+    extendedKeyUsage = serverAuth
+EOF
+}
+
+cmd_selfsign() {
+    # Common variables used by both methods
+    local MACHINE_ID="$(tr -c '[:xdigit:][:space:]' '.' < /etc/machine-id)"
```

Comment:
What does that do? aside from the terminating LF, the file isn't supposed to contain anything other than hex digits anyway (and not even spaces)

Reply:
It's a fairly direct port of the old C code:

```
  if (g_file_get_contents ("/etc/machine-id", &machine_id, NULL, NULL))
    g_strstrip (g_strcanon (machine_id, HEX, ' ')); /* both functions modify machine_id in-place */
```

the variable assignment will absorb the whitespace (which we allow to pass through unmodified, avoiding to change `\n` to `.`, for example).  any non-hex characters are canonicalised...

I clearly misread the original though and saw a `.` there and not a space ` `.  Will adjust that.

Reply:
→ #15608